src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSourceRequest.java

```diff
@@ -321,25 +333,27 @@ public final void setTags(@CheckForNull Iterable<Tag> tags) {
         return Util.fixNull(tags);
     }
 
-
-    // TODO Iterable<Tag> getTags() and setTags(...)
-
     /**
-     * Returns the names of the project collaborators or {@code null} if those details have not been provided yet.
+     * Returns the project {@link Member} or {@code null} if those details have not been provided yet.
      *
-     * @return the names of the project collaborators or {@code null} if those details have not been provided yet.
+     * @return the project {@link Member} or {@code null} if those details have not been provided yet.
      */
-    public final Set<String> getCollaboratorNames() {
-        return collaboratorNames;
+    public final HashMap<String, AccessLevel> getMembers() {
+        return members;
     }
 
     /**
-     * Provides the request with the names of the project collaborators.
+     * Provides the set of project {@link Member}.
      *
-     * @param collaboratorNames the names of the project collaborators.
+     * @param members the set of project {@link Member}.
      */
-    public final void setCollaboratorNames(@CheckForNull Set<String> collaboratorNames) {
-        this.collaboratorNames = collaboratorNames;
+    public final void setMembers(@CheckForNull List<Member> members) {
+        this.members = new HashMap<>();
```

Comment:
I think you can move that up to the field definition
```suggestion
```

Reply:
Do you think it's important? Because this allows me to check if members is null in the `trustedMembers`.

Reply:
Also this complies with the result of the variables in the class which are only initialized in the methods they are set in.

Reply:
it's better to check if members are empty than getting null pointers :laughing:

Reply:
```suggestion
        this.members.clear();
```

Reply:
```java
if (members.isEmpty()) {
do stuff...
} else {
other stuff
}
```